sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggregateExec.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.aggregate
+
+import org.apache.spark.sql.catalyst.expressions.NamedExpression
+import org.apache.spark.sql.catalyst.expressions.aggregate.AggregateExpression
+import org.apache.spark.sql.execution.{ExplainUtils, UnaryExecNode}
+
+/**
+ * Holds common logic for aggregate operators
+ */
+abstract class AggregateExec(
+    groupingExpressions: Seq[NamedExpression],
+    aggregateExpressions: Seq[AggregateExpression])
+  extends UnaryExecNode {
+
+  override def verboseStringWithOperatorId(): String = {
+    val allAggregateExpressions = aggregateExpressions
+
+    val keyString = groupingExpressions.mkString("[", ", ", "]")
+    val functionString = allAggregateExpressions.mkString("[", ", ", "]")
+    val inputString = child.output.mkString("[", ", ", "]")
+    val outputString = output.mkString("[", ", ", "]")
+    s"""
+       |(${ExplainUtils.getOpId(this)}) $nodeName ${ExplainUtils.getCodegenId(this)}
+       |Input: $inputString
+       |Output: $outputString
+       |Keys: $keyString
+       |Functions: $functionString
```

Comment:
instead of only printing the functions, we need to print the full aggregate list, like `a, b + sum(c), sum(d) + max(e)`

Reply:
@cloud-fan 
I updated the HashAggregate testcase for the case you mentioned. The `Output` part is something close to the full aggregate list you mentioned? do you think this is sufficient? 
Just let me know if I misunderstood your suggestion :-)  I can try more.
```
(5) HashAggregate [codegen id : 2]
Input: [count#96L, sum#97L, count#98L]
Keys: []
Functions: [count(val#6), sum(cast(key#5 as bigint)), count(key#5)]
Output: [(count(val) + sum(key))#91L, count(key) FILTER (WHERE (val > 1))#92L]
```

Reply:
can you try `b + sum(c) as abc` and see the output is `abc` or `b + sum(c) as abc`?

Reply:
The column `b` breaks aggregation function check, so I tried with  `sum(c) + 1 as abc`. It only shows alias `abc` but not full aggregate expression. We do need to improve for this case.
```
(5) HashAggregate [codegen id : 2]
Input: [sum#41L, count#42L, count#43L]
Keys: []
Functions: [sum(cast(key#6 as bigint)), count(val#7), count(key#6)]
Output: [abc#26L, (count(val) + sum(key))#36L, count(key) FILTER (WHERE (val > 1))#37L]
```

When removed alias, it shows correctly I think.
```
(5) HashAggregate [codegen id : 2]
Input: [sum#59L, count#60L, count#61L]
Keys: []
Functions: [sum(cast(key#6 as bigint)), count(val#7), count(key#6)]
Output: [(sum(CAST(key AS BIGINT)) + CAST(1 AS BIGINT))#55L, (count(val) + sum(key))#53L, count(key) FILTER (WHERE (val > 1))#54L]
```

Reply:
AFAIK the output only shows the aliases. `sum(c) + 1` works because the default alias is the SQL string.

Reply:
I see, I'll try to add information for full aggregation expressions.

Reply:
@cloud-fan 
`Results` field added in 06e8ef5b465214a3b1821c08918461ab6b0e5a1b to show the full aggregation result expressions, also updated a testcase to test with alias. 
```
(5) HashAggregate [codegen id : 2]
Input: [count#xL, sum#xL, count#xL]
Keys: []
Functions: [count(val#x), sum(cast(key#x as bigint)), count(key#x)]
Results: [(count(val#x)#xL + sum(cast(key#x as bigint))#xL) AS TOTAL#xL, count(key#x)#xL AS count(key) FILTER (WHERE (val > 1))#xL]
Output: [TOTAL#xL, count(key) FILTER (WHERE (val > 1))#xL]
```